### PR TITLE
ISPN-15171 Redesign cross-site commands

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.redhat.com/browse/ISPN-15171

### CODE REVIEW

**Comment by ryanemerson** [comment]
You have an interesting idea of trivial :stuck_out_tongue_winking_eye:

File: core/src/main/java/org/infinispan/xsite/commands/remote/Ids.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package org.infinispan.xsite.commands.remote;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.infinispan.xsite.SingleXSiteRpcCommand;
+
+public final class Ids {
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/xsite/commands/remote/Ids.java:9]: Can be package-private.

File: core/src/main/java/org/infinispan/xsite/commands/remote/IracUpdateKeyRequest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+   }
+
+   @Override
+   protected CompletionStage<T> invokeInLocalCache(String origin, ComponentRegistry registry) {
+      return executeOperation(registry.getBackupReceiver().running());
+   }
+
+   public abstract CompletionStage<T> executeOperation(BackupReceiver receiver);
+
+   public boolean isClear() {
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/xsite/commands/remote/IracUpdateKeyRequest.java:29]: Method is never called.

File: core/src/main/java/org/infinispan/xsite/XSiteCacheMapper.java [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+   private final Map<RemoteCacheInfo, LocalCacheInfoImpl> localCachesMap = new ConcurrentHashMap<>();
+
+   public LocalCacheInfo findLocalCache(String remoteSite, ByteString remoteCacheName) {
+      var key = new RemoteCacheInfo(remoteSite, remoteCacheName);
+      return localCachesMap.computeIfAbsent(key, this::lookupLocalCaches);
+   }
+
+   // for testing only
+   public Optional<LocalCacheInfo> peekLocalCacheForRemoteSite(String remoteSite, ByteString remoteCache) {
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/xsite/XSiteCacheMapper.java:52]: Can be package-private.

File: core/src/main/java/org/infinispan/xsite/commands/remote/XSiteCacheRequest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+   @Override
+   public void writeTo(ObjectOutput output) throws IOException {
+      ByteString.writeObject(output, cacheName);
+      XSiteRequest.super.writeTo(output);
+   }
+
+   @Override
+   public XSiteRequest<T> readFrom(ObjectInput input) throws IOException, ClassNotFoundException {
+      cacheName = ByteString.readObject(input);
+      return XSiteRequest.super.readFrom(input);
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/xsite/commands/remote/XSiteCacheRequest.java:61]: `return this;`?

**Review by ryanemerson — COMMENTED** [review]
The approach seems fine to me, but there are some related test failures on CI.

**Comment by pruivo** [comment]
> The approach seems fine to me, but there are some related test failures on CI.

ops, test was blocking the wrong command :rofl:

**Comment by ryanemerson** [comment]
Thanks @pruivo

**Pull request merged**